xarray/tests/test_indexing.py

```diff
@@ -729,3 +731,16 @@ def test_indexing_1d_object_array() -> None:
     expected = DataArray(expected_data)
 
     assert [actual.data.item()] == [expected.data.item()]
+
+
+def test_indexing_dask_array():
+    da = DataArray(
+        np.ones(10 * 3 * 3).reshape((10, 3, 3)),
+        dims=("time", "x", "y"),
+    ).chunk(dict(time=-1, x=1, y=1))
+    da[{"time": 9}] = 42
+
+    idx = da.argmax("time")
+    actual = da.isel(time=idx)
+
+    assert np.all(actual == 42)
```

Comment:
How about
```suggestion
    with raise_if_dask_computes():
        actual = da.isel(time=dask.array.from_array([9], chunks=(1,))
    expected = da.isel(time=[9])
    assert_identical(expected, actual)
```
with `from . import raise_if_dask_computes` at the top.

We could test `9`, `[9]`, and `[9, 9]`

Reply:
When the indexer is an independent Dask array, the computation is indeed not triggered. 
However, when the indexer is constructed from `da` it does trigger the computation, even if the dimensions do not have coordinate variable.
That's why the unit test uses `idx = da.argmax("time")`.

Testing both would make sense though.

Reply:
> I thought it would only be triggered when the indexer is constructed from da.

Yeah that `raise_if_dask_computes` decorator is great for checking these ideas.

I've got it fixed but we should wait for @benbovy to finish with #5692 (though it doesn't look like that PR conflicts with these changes). It would be useful to add more test cases here.

Reply:
Some other tests could be the ones in https://github.com/pydata/xarray/issues/4276, https://github.com/pydata/xarray/issues/4663, and this one from #2511

``` python
import dask.array as da
import numpy as np
import xarray as xr
from xarray.tests import raise_if_dask_computes

darr = xr.DataArray(data=[0.2, 0.4, 0.6], coords={"z": range(3)}, dims=("z",))
indexer = xr.DataArray(
    data=np.random.randint(0, 3, 8).reshape(4, 2).astype(int),
    coords={"y": range(4), "x": range(2)},
    dims=("y", "x"),
)
with raise_if_dask_computes():
    actual = darr[indexer.chunk({"y": 2})]

xr.testing.assert_identical(actual, darr[indexer])
```

Which actually does work now, but does end up computing

Reply:
TBH, I was not expecting these changes to be merged, it was only to support the discussion on the issue but it's nice if can be integrated.
I'll try to add the unit tests.